File: docs/code_snippet/java/checkoutUrlPayment.java
new ChapaCheckoutPage(this).processPayment("chapa-checkout-url", new ChapaCheckoutUrlCallback() {

    /**
     * Called when payment is successful
     */
    @Override
    public void onSuccess() {
        // TODO your code here
    }

    /**
     * Called when error occurred in payment
     */
    @Override
    public void onError(@NonNull ChapaError chapaError) {
        // TODO your code here
    }

    /**
     * Called when payment is canceled
     */
    @Override
    public void onCancel() {
        // TODO your code here

    }
});

File: docs/code_snippet/kotlin/checkoutUrlPayment.kt
ChapaCheckoutPage(this).processPayment("chapa-checkout-url",object : ChapaCheckoutUrlCallback {

 /**
  * Called when payment is successful
  */
 override fun onSuccess() {
     // TODO your code here
 }

 /**
  * Called when error occurred in payment
  */
 override fun onError(chapaError: ChapaError) {
     // TODO your code here
 }

 /**
  * Called when payment is canceled
  */
 override fun onCancel() {
     // TODO your code here
 }
})